phac-nml/irida
# LDAP and ADLAP backend changes

## Description of changes
Backend changes for LDAP and ADLDAP support

## Related issue
#227 

### How to set up LDAP to test
 1. Download (https://directory.apache.org/studio/download/download-linux.html) and setup an ldap server with apache directory studio, this guide explains it step by step to get basic users https://docs.craftercms.org/en/3.0/developers/cook-books/how-tos/setting-up-an-ldap-server-for-dev.html
 2. Start IRIDA (no ldap) and create a user account that exists in your ldap server (ex. jwick)
 3. Stop IRIDA, then configure your authentication.configuration file, below is an an example that should work with the above guide
```
# Alternative Authentication modes: ldap, adldap
# Defaults is in memory authentication with bcrypt
irida.administrative.authenitcation.mode=ldap

# LDAP config
irida.administrative.authenitcation.ldap.url=ldap://localhost:10389/
irida.administrative.authentication.ldap.base=dc=example,dc=com
irida.administrative.authenitcation.ldap.userdn=uid=admin,ou=system
irida.administrative.authenitcation.ldap.password=secret
irida.administrative.authentication.ldap.userdn_search_patterns=cn={0},ou=Users
# Optional Argument. change this to 'follow' to handle referrals. See AbstractContextSource javadocs for more info
irida.administrative.authenitcation.ldap.set_referral=ignore
```
 4. On Directory Studio, change jwick's password to something you know, like `password`
 5. Start IRIDA, and sign in with the user jwick

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:
